### PR TITLE
Add MPI support

### PR DESCRIPTION
**Pull request by pabloferz**
I had once started writing a LAMMPS wrapper over https://github.com/pabloferz/LAMMPSWrapper.jl. So I thought I could move some bits from there to here.

@vchuravy

### CODE REVIEW

**Comment by vchuravy** [comment]
Thanks! I am waiting for https://github.com/JuliaParallel/MPI.jl/pull/513 to land.

File: src/LAMMPS.jl [hunk 1]
```diff
@@ -50,6 +57,7 @@ function check(lmp::LMP)
 end
 
 function command(lmp::LMP, cmd)
+    check_init(lmp)
```
Comment (vchuravy) [inline src/LAMMPS.jl:60]: Why do we need these calls to `check_init`? We should error on constructing an incorrect LAMMPS instance instead.

File: src/LAMMPS.jl [hunk 2]
```diff
@@ -250,4 +269,11 @@ function gather_atoms(lmp::LMP, name, T, count)
     return data
 end
 
+function __init__()
+    if !MPI.Initialized()
```
Comment (vchuravy) [inline src/LAMMPS.jl:273]: We should do this as the library for the user. The user might want to change the `threadlevel`

cc: @simonbyrne

File: src/api.jl [hunk 3]
```diff
@@ -1,9 +1,16 @@
 module API
 
 using CEnum
+using MPI
```
Comment (vchuravy) [inline src/api.jl:4]: This File is auto-generated, can we do this automatically in `res/wrap.jl`?

File: src/LAMMPS.jl [hunk 4]
```diff
@@ -1,29 +1,28 @@
...
-    function LMP(args::Vector{String}=String[])
-        if isempty(args)
-            argsv = C_NULL
-        else
-            args = copy(args)
-            pushfirst!(args, "lammps")
-            argsv = map(pointer, args)
-        end
+    function LMP(args::Vector{String}=String[], comm = MPI.COMM_WORLD)
+        ptrref = Ref{LAMMPSPtr}(C_NULL)
```
Comment (vchuravy) [inline src/LAMMPS.jl:17]: I think you still need to `pushfirst!(args, "lammps")`. IIRC lammps expects the binary name as the first arg.

File: src/LAMMPS.jl (hunk 1 above)
Comment (pabloferz) [reply src/LAMMPS.jl:60]: Fair enough. Will make sure the constructor enforces proper LAMMPS instances

**Comment by vchuravy** [comment]
@pabloferz thank you for the kick in the butt :) I am going to work on this in #21

**Pull request closed without merging**